### PR TITLE
export wallet

### PR DESCRIPTION
**Pull request by EndNationStates**
Added wallet export in plain text and CC supported json #44

### CODE REVIEW

File: src/apps/xpubs/xpubs.py [hunk 1]
```diff
@@ -69,10 +73,59 @@ async def menu(self, show_screen, show_all=False):
...
+                        "account": self.account,
+                        "chain": "BTC" if self.network == "main" else "XTN"
+                    }
+
+                    for der in derivations:
+                        data[der[0]] = {
+                            "name": der[1],
+                            "deriv": der[2],
+                            "xpub": self.calculate_xpub(der[2]),
+                            "_pub": self.calculate_xpub(der[2], True),
```
Comment (stepansnigirev) [inline src/apps/xpubs/xpubs.py:111]: Calculating xpubs is an expensive function (it does some EC math in `keystore.get_xpub`),
so better to call `keystore.get_xpub` once and then serialize it to xpub and slip132.
I'd suggest deriving the key in this function once per derivation path, and then serializing it with proper version.

Comment (EndNationStates) [reply src/apps/xpubs/xpubs.py:111]: Will rewrite

File: src/apps/xpubs/xpubs.py [hunk 2]
```diff
@@ -123,19 +179,38 @@ async def show_xpub(self, derivation, show_screen):
...
+    def write_file(self, filename, filedata):
+        if not platform.is_sd_present():
+            raise AppError("SD card is not present")
+        platform.mount_sdcard()
+        with open(platform.fpath("/sd/%s" % filename), "w") as f:
+            f.write(filedata)
+        platform.unmount_sdcard()
+
+    async def save_menu(self, show_screen):
+        buttons = [(0, "Specter-DIY"), (1, "Cold Card")]
```
Comment (stepansnigirev) [inline src/apps/xpubs/xpubs.py:199]:
```suggestion
        buttons = [(0, "Specter-DIY (plaintext)"), (1, "Cold Card (json)")]
```

File: src/apps/xpubs/xpubs.py [hunk 3]
```diff
@@ -69,10 +73,62 @@ async def menu(self, show_screen, show_all=False):
...
+                    ('bip44', "p2pkh", "m/44'/%d'/%d'" % (coin, self.account)),
+                    ('bip48_2', "p2wsh", "m/48'/%d'/%d'/2'" % (coin, self.account)),
+                    ('bip48_1', "p2sh-p2wsh", "m/48'/%d'/%d'/1'" % (coin, self.account)),
+                ]
+
+                if format == self.export_specter_diy:
+                    filedata = ""
+                    for der in derivations:
+                        xpub = self.keystore.get_xpub(der[2])
+                        filedata += "[%s/%s]%s\n" % (fingerprint, der[2], xpub.to_base58(net["xpub"]))
```
Comment (stepansnigirev) [inline src/apps/xpubs/xpubs.py:96]: Right now it writes `[fingerprint/m/deriavtion]xpub`, shoudl be `[fingerprint/derivation]xpub`. Also we noramlly use `h` for hardened derivation. We should use `'` only for compatibility with coldcard in json files.
```suggestion
                        filedata += "[%s/%s]%s\n" % (fingerprint, der[2][2:].replace("'","h"), xpub.to_base58(net["xpub"]))
```

File: src/apps/xpubs/xpubs.py [hunk 4]
```diff
@@ -52,7 +55,8 @@ async def menu(self, show_screen, show_all=False):
                 ),
             ]
         else:
-            buttons += [(0, "Show more keys"), (2, "Change account number"), (1, "Enter custom derivation")]
+            buttons += [(0, "Show more keys"), (2, "Change account number"), (1, "Enter custom derivation"),
+                        (3, "Export all keys")]
```
Comment (stepansnigirev) [inline src/apps/xpubs/xpubs.py:59]:
```suggestion
                        (3, "Export all keys to SD")]
```

**Review by stepansnigirev — COMMENTED** [review]
Last few things, otherwise looks good.

**Pull request merged**